src/config/globalInfoConfig.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { isEmpty } from '@salesforce/kit';
+import { JsonMap } from '@salesforce/ts-types';
+import { SfAuthorization } from '../authInfo';
+import { Global } from '../global';
+import { ConfigFile } from './configFile';
+import { ConfigContents, ConfigValue } from './configStore';
+import { SfdxDataHandler } from './sfdxDataHandler';
+
+export enum SfDataKeys {
+  AUTHORIZATIONS = 'authorizations',
+}
+
+type Timestamp = { timestamp: Date };
+
+export interface Authorizations {
+  [key: string]: SfAuthorization & Timestamp;
+}
+
+export type SfData = {
+  [SfDataKeys.AUTHORIZATIONS]: Authorizations;
+};
+
+export class GlobalInfo extends ConfigFile<ConfigFile.Options> {
+  public static EMPTY_DATA_MODEL: SfData = {
+    [SfDataKeys.AUTHORIZATIONS]: {},
+  };
+  private static instance: GlobalInfo;
+  private enableInteroperability = true;
+  private sfdxHandler = new SfdxDataHandler();
+
+  public static async getInstance(): Promise<GlobalInfo> {
+    if (!GlobalInfo.instance) {
+      GlobalInfo.instance = await GlobalInfo.create(GlobalInfo.getOptions());
+    }
+    return GlobalInfo.instance;
+  }
+
+  public static getFileName(): string {
+    return 'sf.json';
+  }
+
+  /**
+   * Gets default options for the SfConfig
+   */
+  public static getOptions(): ConfigFile.Options {
+    return {
+      isGlobal: true,
+      isState: true,
+      filename: GlobalInfo.getFileName(),
+      stateFolder: Global.SF_STATE_FOLDER,
+    };
+  }
+
+  public getContents(): SfData {
+    return this['contents'] as SfData;
+  }
+
+  public get authorizations(): Authorizations {
+    return this.getContents()[SfDataKeys.AUTHORIZATIONS];
+  }
+
+  public getAuthorization(username: string): SfAuthorization {
+    return this.getContents()[SfDataKeys.AUTHORIZATIONS][username];
+  }
+
+  public setAuthorization(username: string, authorization: SfAuthorization): void {
+    this.set(`${SfDataKeys.AUTHORIZATIONS}["${username}"]`, authorization);
+  }
+
+  public unsetAuthorization(username: string): void {
+    this.unset(`${SfDataKeys.AUTHORIZATIONS}["${username}"]`);
+  }
+
+  public set<U extends JsonMap>(key: string, value: U): ConfigContents {
+    return super.set(key, this.timestamp(value));
+  }
+
+  public async write(newContents?: SfData): Promise<SfData> {
+    const result = (await super.write(newContents)) as SfData;
+    if (this.enableInteroperability) await this.sfdxHandler.write(result);
+    return result;
+  }
+
+  protected async init(): Promise<void> {
+    const contents = this.enableInteroperability ? await this.mergeWithSfdxData() : await this.loadSfData();
```

Comment:
Ahh I see. This would be awesome if just turning this off made all the unit tests pass =) but I see how the mocks will probably cause problems.

 We should also profile this init with interoperability to make sure it is fine. I don't think there is a performance hit but good to make sure.